src/contracts/FoxStaking.sol

```diff
@@ -354,12 +380,27 @@ contract FoxStaking is Ownable {
         @param _amount uint
         @param _trigger bool
      */
-    function unstake(uint256 _amount, bool _trigger) external {
+    function unstake(uint256 _amount, bool _trigger) external returns (bool) {
         if (_trigger) {
             rebase();
         }
+
         IERC20(FOXy).safeTransferFrom(msg.sender, address(this), _amount);
-        // TODO: withdraw from TOKE and create cooldown claim
+
+        Claim memory info = cooldownInfo[msg.sender];
+        require(!info.lock, "Deposits for account are locked");
+
+        cooldownInfo[msg.sender] = Claim({
+            amount: info.amount.add(_amount),
```

Comment:
This is probably a larger issue to tackle but we should do the following 

1. remove all of the hardcoded imports and use libraries (this was just a vestige of the forked code) and I think they did it for etherscan verification purposes, but we should avoid it if possible.
2. Upgrade to solidity 0.8.4 (or more recent) which has safe math built in
3. get ride of all of the unneeded safemath.

Reply:
I'll make a ticket for this and push it up in a separate PR today or tomorrow